### PR TITLE
[DEV-1242, DEV-1206, DEV-1300] EVM bridge integrations, LM Exchange integration, LM external API

### PR DESCRIPTION
**Pull request by TonySemikin**
#### Pre-Release

- [x] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [x] Check for linter errors (in PR)
- [x] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

File: src/integration/blockchain/arbitrum/arbitrum-client.ts [hunk 1]
```diff
@@ -23,4 +183,55 @@ export class ArbitrumClient extends EvmClient {
...
+
+    this.nonce.set(fromAddress, nonce + 1);
+
+    return tx.hash;
+  }
+
+  /**
+   * @TODO
+   * consider using this as a primary source of estimating gas in super class
+   */
```
Comment (TonySemikin) [inline src/integration/blockchain/arbitrum/arbitrum-client.ts:217]: kept TODO on purpose

File: src/subdomains/core/liquidity-management/services/liquidity-management.service.ts [hunk 2]
```diff
@@ -38,8 +40,42 @@ export class LiquidityManagementService {
...
+    const rule = await this.findRuleByAssetOrThrow(assetId);
+
+    if (!rule.redundancyStartAction) {
+      throw new BadRequestException(`Rule ${rule.id} does not support liquidity redundancy path`);
+    }
+
+    const liquidityState = { deficit: 0, redundancy: amount };
+
+    return this.executeRule(rule, liquidityState);
+  }
```
Comment (davidleomay) [inline src/subdomains/core/liquidity-management/services/liquidity-management.service.ts:67]: Should we check for current liquidity state and combine both inputs?

File: src/subdomains/supporting/dex/services/dex-bitcoin.service.ts [hunk 3]
```diff
@@ -10,17 +12,44 @@ import { LiquidityOrderRepository } from '../repositories/liquidity-order.reposi
...
   }
 
+  async sendUtxoToMany(payout: { addressTo: string; amount: number }[]): Promise<string> {
+    const feeRate = await this.feeService.getRecommendedFeeRate();
+    return this.#client.sendMany(payout, feeRate);
+  }
+
+  async transferMinimalUtxo(address: string): Promise<string> {
+    return this.sendUtxoToMany([{ addressTo: address, amount: Config.blockchain.default.minDeposit.Bitcoin.BTC / 2 }]);
+  }
```
Comment (davidleomay) [inline src/subdomains/supporting/dex/services/dex-bitcoin.service.ts:30]: Is this used? `Config.blockchain.default.minDeposit.Bitcoin.BTC / 2` is currently more than 5 USD 😉

File: src/integration/blockchain/arbitrum/arbitrum-client.ts [hunk 4]
```diff
@@ -14,6 +36,144 @@ export class ArbitrumClient extends EvmClient {
...
+    });
+
+    await approveTx.wait();
+
+    const depositTx = await erc20Bridge.deposit({
+      amount: this.convertToWeiLikeDenomination(amount, decimals),
+      erc20L1Address: l1Token.chainId,
+      l1Signer: this.#l1Wallet,
+      l2Provider: this.provider,
+    });
```
Comment (davidleomay) [inline src/integration/blockchain/arbitrum/arbitrum-client.ts:93]: How long does this approval take? What if the service restarts at this point?

File: src/subdomains/core/liquidity-management/entities/liquidity-management-pipeline.entity.ts [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
 import { LiquidityManagementOrderStatus, LiquidityManagementPipelineStatus, LiquidityOptimizationType } from '../enums';
-import { LiquidityVerificationResult } from '../interfaces';
+import { LiquidityState } from '../interfaces';
 import { LiquidityManagementAction } from './liquidity-management-action.entity';
 
 @Entity()
+@Index((p: LiquidityManagementPipeline) => [p.rule, p.status], {
+  unique: true,
+  where: `status IN ('${LiquidityManagementPipelineStatus.CREATED}', '${LiquidityManagementPipelineStatus.IN_PROGRESS}')`,
+})
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/subdomains/core/liquidity-management/entities/liquidity-management-pipeline.entity.ts:12]: remove the p.status from index

File: src/subdomains/core/liquidity-management/services/liquidity-management.service.ts (hunk 2 above)
Comment (TonySemikin) [reply src/subdomains/core/liquidity-management/services/liquidity-management.service.ts:67]: lets keep it like that

File: src/subdomains/supporting/dex/services/dex-bitcoin.service.ts (hunk 3 above)
Comment (TonySemikin) [reply src/subdomains/supporting/dex/services/dex-bitcoin.service.ts:30]: / 1000 instead

File: src/subdomains/supporting/dex/interfaces/index.ts [hunk 6]
```diff
@@ -26,6 +26,17 @@ export interface LiquidityRequestOptions {
   estimatePurchaseFee?: boolean;
 }
 
+export interface TransactionQuery {
+  asset: Asset;
+  amount: number;
+  since: Date;
+}
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/subdomains/supporting/dex/interfaces/index.ts:33]: as discussed - this is not needed

File: src/integration/exchange/services/exchange.service.ts [hunk 7]
```diff
@@ -18,6 +19,10 @@ enum OrderStatus {
   CANCELED = 'canceled',
 }
 
+export type ExchangeExtended = Exchange & { depositAddresses: { [key in Blockchain]?: string } } & {
+  withdrawalAddressesKeys: { [key in Blockchain]?: string };
+};
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/integration/exchange/services/exchange.service.ts:24]: removed

File: src/integration/blockchain/arbitrum/arbitrum-client.ts (hunk 4 above)
Comment (TonySemikin) [reply src/integration/blockchain/arbitrum/arbitrum-client.ts:93]: as discussed, in the first approach lets keep it like this, ideally to be split into separate actions

File: src/integration/blockchain/arbitrum/arbitrum-client.ts [hunk 8]
```diff
@@ -23,4 +183,55 @@ export class ArbitrumClient extends EvmClient {
...
+   * @TODO
+   * consider using this as a primary source of estimating gas in super class
+   */
+  private async getCurrentGasForCoinTransaction(amount: number): Promise<BigNumber> {
+    return this.provider.estimateGas({
+      from: this.dfxAddress,
+      to: this.randomReceiverAddress,
+      value: this.convertToWeiLikeDenomination(amount, 'ether'),
+    });
+  }
```
[the remark this answers is not in the document]
Comment (TonySemikin) [reply src/integration/blockchain/arbitrum/arbitrum-client.ts:224]: you are right, `getCurrentGasForCoinTransaction` can be reused in `getCurrentGasCostForCoinTransaction`

**Review by davidleomay — APPROVED** [review]
Need to update DEV environment

**Pull request merged**